src/jdk.compiler/share/classes/com/sun/tools/javac/parser/DocCommentParser.java

```diff
@@ -311,33 +310,29 @@ protected DCTree inlineTag() {
         int p = bp - 1;
         try {
             nextChar();
-            if (isIdentifierStart(ch)) {
-                Name name = readTagName();
-                TagParser tp = tagParsers.get(name);
-
-                if (tp == null) {
+            if (!isIdentifierStart(ch)) {
+                return erroneous("dc.no.tag.name", p);
+            }
+            Name name = readTagName();
+            TagParser tp = tagParsers.get(name);
+            if (tp == null) {
+                skipWhitespace();
+                DCTree text = inlineText(WhitespaceRetentionPolicy.REMOVE_ALL);
+                nextChar();
+                return m.at(p).newUnknownInlineTagTree(name, List.of(text)).setEndPos(bp);
+            } else {
+                if (!tp.retainWhiteSpace) {
                     skipWhitespace();
-                    DCTree text = inlineText(WhitespaceRetentionPolicy.REMOVE_ALL);
-                    if (text != null) {
-                        nextChar();
-                        return m.at(p).newUnknownInlineTagTree(name, List.of(text)).setEndPos(bp);
-                    }
-                } else {
-                    if (!tp.retainWhiteSpace) {
-                        skipWhitespace();
-                    }
-                    if (tp.getKind() == TagParser.Kind.INLINE) {
-                        DCEndPosTree<?> tree = (DCEndPosTree<?>) tp.parse(p);
-                        if (tree != null) {
-                            return tree.setEndPos(bp);
-                        }
-                    } else { // handle block tags (for example, @see) in inline content
-                        inlineText(WhitespaceRetentionPolicy.REMOVE_ALL); // skip content
-                        nextChar();
-                    }
+                }
+                if (tp.allowsInline()) {
+                    DCEndPosTree<?> tree = (DCEndPosTree<?>) tp.parse(p, TagParser.Kind.INLINE);
+                    return tree.setEndPos(bp);
+                } else { // handle block tags (for example, @see) in inline content
+                    DCTree text = inlineText(WhitespaceRetentionPolicy.REMOVE_ALL); // skip content
+                    nextChar();
+                    return m.at(p).newUnknownInlineTagTree(name, List.of(text)).setEndPos(bp);
```

Comment:
The code is correct, and even the comment: which refers to "block tags in inline content".
The case being handled is one like this:

````
    /**
     * This is a bad comment. {@see Object} or {@since 7}.
     */
````